src/com/googlecode/jmxtrans/model/naming/KeyUtils.java

```diff
@@ -160,7 +160,7 @@ public static String getConcatedTypeNameValues(List<String> typeNames, String ty
 		StringBuilder sb = new StringBuilder();
 		for (String key : typeNames) {
 			String result = typeNameValueMap.get(key);
-			if (result != null) {
+			if (result != null && !result.isEmpty()) {
```

Comment:
Also it seem to me that this whole block could be written in a much simpler way by using the functional idioms of Guava.